### PR TITLE
Stop overwriting aggregations on same column

### LINKED ISSUE #655: [BUG][Datafusion] Queries with multiple distinct aggregations return incorrect results
**What happened**:
Queries that include multiple distinct aggregations on the same column (ex: `SUM(a)` and `AVG(a)`) return incorrect results.

**Minimal Complete Verifiable Example**:

```python
import pandas as pd
from dask_sql import Context

c = Context()

df = pd.DataFrame({"a":[1,2,4]})
c.create_table("df", df)

c.sql("SELECT SUM(a), AVG(a) FROM df").compute()
```
returns:
```
   SUM(df.a)  AVG(df.a)
0        2.0        2.0
```
and:
```python
import pandas as pd
from dask_sql import Context

c = Context()

df = pd.DataFrame({"a":[1,2,4]})
c.create_table("df", df)

c.sql("SELECT AVG(a), SUM(a) FROM df").compute()
```
returns:
```
   AVG(df.a)  SUM(df.a)
0          7          7
```

**Environment**:
- dask-sql version: 8/1 datafusion build
- Python version: 3.9
- Operating System: Ubuntu 18.04.4
- Install method (conda, pip, source): source

### PR DESCRIPTION
**Pull request by ChrisJar**
Closes #655

### CODE REVIEW

File: dask_sql/physical/rel/logical/aggregate.py [hunk 1]
```diff
@@ -185,9 +185,17 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
         if len(output_column_order) == 1 and output_column_order[0] == "UInt8(1)":
             backend_output_column_order = [df_agg.columns[0]]
         else:
+
+            def try_get_backend_by_frontend_name(oc):
+                try:
+                    return cc.get_backend_by_frontend_name(oc)
+                except KeyError:
+                    return oc
+
```
Comment (ayushdg) [inline dask_sql/physical/rel/logical/aggregate.py:194]: do you think this is worth moving to the column_container class? or do you think this is a one off case needed only here?

File: dask_sql/physical/rel/logical/aggregate.py [hunk 2]
```diff
@@ -370,7 +378,7 @@ def _collect_aggregations(
                 )
 
             # Finally, extract the output column name
-            output_col = str(inputs[0].column_name(rel))
+            output_col = expr.toString()
```
Comment (ayushdg) [inline dask_sql/physical/rel/logical/aggregate.py:381]: Out of curiosity, to any of the PyExpr methods like `get_column_name` return the correct results, or do we have to rely on the stringified name for the Expr?

**Review by ayushdg — CHANGES_REQUESTED** [review]
Could you add a test with the original issue highlighting this (or uncomment any existing one if being skipped) that tests for these cases

File: dask_sql/physical/rel/logical/aggregate.py (hunk 2 above)
Comment (ChrisJar) [reply dask_sql/physical/rel/logical/aggregate.py:381]: Yeah the `column_name` function just returns the input column

File: dask_sql/physical/rel/logical/aggregate.py (hunk 1 above)
Comment (ChrisJar) [reply dask_sql/physical/rel/logical/aggregate.py:194]: I think this is a pretty one off case

**Review by ayushdg — APPROVED** [review]
changes lgtm! @charlesbluca could you also take a look, since you're somewhat familiar with this part of the codebase

**Pull request merged**